cdci_data_analysis/flask_app/app.py

```diff
@@ -221,6 +227,26 @@ def common_exception_payload():
     return payload
 
 
+
+class ExitStatus(Schema):
+    status = fields.Int(validate=OneOf([0, 1]))
+    message = fields.Str(desciption="if query_status == 'failed', shown in waitingDialog in red")     
+    error_message = fields.Str(desciption="if query_status == 'failed', shown in waitingDialog in red")     
+    debug_message = fields.Str(desciption="if query_status == 'done' but exit_status.status != 0, shown in waitingDialog in red")     
+    comment = fields.Str(desciption="always, shown in waitingDialog in yellow")     
+    warning = fields.Str(desciption="")     
+    
+
+class QueryOutJSON(Schema):
+    query_status = fields.Str(
+                        validate=OneOf(["done", "failed"]),
```

Comment:
`submitted` status not considered?

Reply:
I add, but it does not really matter, since it is not currently exhaustive: only parts needed for current cause - reporting on frontend - are described.